### PR TITLE
upgrade to react 17 and ts 4

### PR DESCRIPTION
**Pull request by guerrap**
# Summary

Upgrade the project to use react `17.0.2` and typescript `4.1.2`.

Done:
- [x] Rename all components files to `*.jsx`
- [x] Remove `React` imports
- [x] Fix an issue with `react-error-overlay` 

# How To Test

1.  Launch dxstats
- [x] The application should still work the same

### CODE REVIEW

**Review by wixzi — APPROVED** [review]
Awesome

**Comment by wixzi** [comment]
Screenshot 2022-07-04 at 23 09 31
This background is little too similar to link color. Maybe we can make it similar to other pages (darker shade)

**Comment by guerrap** [comment]
> Screenshot 2022-07-04 at 23 09 31 This background is little too similar to link color. Maybe we can make it similar to other pages (darker shade)

FIxed this in the PR #87

File: src/components/Search/index.jsx [hunk 1]
```diff
@@ -235,16 +235,21 @@ export default function Search({ small = false }) {
...
+    let found = {};
+
+    allTokens &&
+      allTokens.map((token) => {
+        if (!found[token.id]) {
+          found[token.id] = true;
+          tokens.push(token);
+        }
+        return true;
+      });
```
Comment (wixzi) [inline src/components/Search/index.jsx:249]: option 1

```suggestion
      allTokens?.forEach((token) => {
        if (!found[token.id]) {
          found[token.id] = true;
          tokens.push(token);
        }
      });
```

suggested option 2 

```suggestion
const tokens = [...new Map(allTokens.map(token =>
  [token.id, token])).values()]
```

File: src/components/Search/index.jsx [hunk 2]
```diff
@@ -259,16 +264,21 @@ export default function Search({ small = false }) {
...
+      allPairs.map((pair) => {
+        if (!found[pair.id]) {
+          found[pair.id] = true;
+          pairs.push(pair);
+        }
+        return true;
+      });
+
+    return pairs;
+  }, [allPairs]);
```
Comment (wixzi) [inline src/components/Search/index.jsx:281]:
```suggestion  
   const uniquePairs = useMemo(() => {
return [...new Map(allPairs?.map(pair =>
  [pair.id, pair])).values()]
  }, [allPairs]);
```

**Review by wixzi — APPROVED** [review]
Looks good 👍🏼

**Pull request merged**